publishing Components (#621)
ml6team/fondant: docs/components/publishing_components.md

updated, ensuring that the next pipeline run correctly references the image.

!!! note "IMPORTANT"   

    When developing custom components using the local runner, building and publishing components is not required.

>     That is because components that are not located in the registry (local custom components) will be built during runtime when using the local runner. 

Reviewer: ```suggestion
    That is because components that are not located in the registry (local custom components) will be built automatically by the local runner. 
```
